### PR TITLE
Feature/add publish for removing images

### PR DESCRIPTION
**Pull request by SolovyovDmitry**

### CODE REVIEW

File: src/EducationService.Business/Commands/Images/RemoveImagesCommand.cs [hunk 1]
```diff
@@ -110,9 +76,16 @@ public async Task<OperationResultResponse<bool>> ExecuteAsync(RemoveImagesReques
...
       {
-        await RemoveAsync(request.ImagesIds, response.Errors);
+        return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.BadRequest);
       }
 
+      await Task.WhenAll(
+        request.ImagesIds.Count > 0
+          ? _publish.RemoveImagesAsync(
+            imageIds: request.ImagesIds,
+            imageSource: ImageSource.User)
```
Comment (SolovyovDmitry) [inline src/EducationService.Business/Commands/Images/RemoveImagesCommand.cs]: Seems like I am supposed to update ImageSource enum

File: src/EducationService.Broker/Publishes/Interfaces/IPublish.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.EducationService.Broker.Publishes.Interfaces
+{
+  [AutoInject]
+  public interface IPublish
+  {
+    Task RemoveImagesAsync(List<Guid> imageIds, ImageSource imageSource);
```
Comment (VLiamin) [inline src/EducationService.Broker/Publishes/Interfaces/IPublish.cs]: imagesIds

File: src/EducationService.Broker/Publishes/Publish.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  {
+    private readonly IBus _bus;
+
+    public Publish(
+      IBus bus)
+    {
+      _bus = bus;
+    }
+
+    public async Task RemoveImagesAsync(List<Guid> imageIds, ImageSource imageSource)
```
Comment (VLiamin) [inline src/EducationService.Broker/Publishes/Publish.cs]: imagesIds

**Review by VLiamin — APPROVED** [review]

File: src/EducationService.Broker/Publishes/Publish.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  {
+    private readonly IBus _bus;
+
+    public Publish(
+      IBus bus)
+    {
+      _bus = bus;
+    }
+
+    public async Task RemoveImagesAsync(List<Guid> imagesIds, ImageSource imageSource)
```
Comment (MariyaSin) [inline src/EducationService.Broker/Publishes/Publish.cs]: the service use one imageSourse, you do not have to accept this parameter in the arguments

File: src/EducationService.Broker/Publishes/Publish.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    {
+      await _bus.Publish<IRemoveImagesPublish>(IRemoveImagesPublish.CreateObj(
+        imagesIds: imagesIds,
+        imageSource: imageSource));
```
Comment (MariyaSin) [inline src/EducationService.Broker/Publishes/Publish.cs]: imageSourse.Education

File: src/EducationService.Models.Dto/Configurations/RabbitMqConfig.cs [hunk 6]
```diff
@@ -14,13 +15,13 @@ public class RabbitMqConfig : BaseRabbitMqConfig
...
     // image
-    [AutoInjectRequest(typeof(ICreateImagesRequest))]
+    [AutoInjectRequest(typeof(ICreateImagesPublish))]
     public string CreateImagesEndpoint { get; set; }
 
     [AutoInjectRequest(typeof(IGetImagesRequest))]
     public string GetImagesEndpoint { get; set; }
 
-    [AutoInjectRequest(typeof(IRemoveImagesRequest))]
+    [AutoInjectRequest(typeof(IRemoveImagesPublish))]
```
Comment (MariyaSin) [inline src/EducationService.Models.Dto/Configurations/RabbitMqConfig.cs]: do you need it?

File: src/EducationService.Business/Commands/Images/RemoveImagesCommand.cs [hunk 7]
```diff
@@ -28,61 +25,30 @@ public class RemoveImagesCommand : IRemoveImagesCommand
     private readonly IImageRepository _imageRepository;
     private readonly IEducationRepository _educationRepository;
     private readonly IHttpContextAccessor _httpContextAccessor;
-    private readonly IRequestClient<IRemoveImagesRequest> _rcRemoveImages;
```
Comment (MariyaSin) [inline src/EducationService.Business/Commands/Images/RemoveImagesCommand.cs:31]: remove the endpoint from upseting.json and rabbitMqConfig

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**